api/handlers/filetypehandler.py

```diff
@@ -0,0 +1,38 @@
+from ..web import base
+from .. import config
+from .. import validators
+from ..auth import userauth
+from ..dao import noop
+
+class FileType(base.RequestHandler):
+
+    def get(self):
+        """Get file types"""
+        return config.db.filetypes.find()
+
+    def post(self):
+        """
+        Insert or replace a file type. Required fields: '_id' and 'regex' where the '_id' is the unique name of
+        the file type and 'regex' is a regular expression which is used to figure out the file type from the file name.
+        """
+        permchecker = userauth.default(self)
+        payload = self.request.json_body
+        mongo_schema_uri = validators.schema_uri('mongo', 'filetype.json')
+        mongo_validator = validators.decorator_from_schema_path(mongo_schema_uri)
+        mongo_validator(permchecker(noop))('PUT', payload=payload)
```

Comment:
We can also use a mongo schema validator (although we've kind of stopped that practice as we didn't see many real world situations where it prevented dev error), but this is better labeled as an input validator as it's using a json schema to validate request bodies from clients. Rather than creating a decorator*, you can call the validator directly ([here](https://github.com/scitran/core/blob/master/api/jobs/handlers.py#L230) is an example). 

Related, when you call the existing validator with `PUT`, it drops the `required` piece of the json schema to allow for patch updates. `POST` does not. 

\* This decorator practice was created in the old rewrite of containerhandler to facilitate calling the permissions check, validators and storage function in one line. It works _alright_ in that handler for now, but new functionality does not need to follow that convention.